### PR TITLE
Provide advice for "no peer available" error

### PR DESCRIPTION
**Pull request by kriskowal**
r @jcorbin @Raynos @ShanniLi

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -49,6 +49,12 @@ Logger.prototype.error = function error(err) {
         self.exitCode = self.exitCode | err.errorCode & 0x7f;
 
         console.error(err.name + ': ' + err.message);
+
+        if (err.message.lastIndexOf('no peer available', 0) === 0) {
+            console.error('With Hyperbahn, this likely means that the service you are trying to reach is not advertising.');
+            console.error('Without Hyperbahn, this likely means that the peer you are trying to reach is not listening.');
```
Comment (ShanniLi) [inline logger.js]: I remembered the second case should be a connection error, right?

Comment (ShanniLi) [inline logger.js]: maybe adding a test case for the second scenario?

Comment (Raynos) [inline logger.js]: Agreed second scenario would only happen if `-H /host/file` was an empty array. I think there is no "no peer available" outside of Hyperbahn.

**Comment by kriskowal** [comment]
Addressed feedback. Intend to merge.

**Comment by ShanniLi** [comment]
lgtm

**Pull request merged**